src/frame_buffer/frame_buffer.rs

```diff
@@ -1,9 +1,13 @@
-use core::any::Any;
 use core::fmt::{Debug, Formatter};
 use core::ops::{Deref, DerefMut};
 
-pub type DropFunction = fn(&mut FrameBuffer, DropMetadata);
-pub type DropMetadata = &'static (dyn Any + Send + Sync);
+use crate::frame_buffer::single_frame_allocator::SingleFrameAllocator;
+
+pub enum FrameAllocators<'a> {
+    #[cfg(test)]
+    Dummy,
+    SingleFrameAllocator(&'a SingleFrameAllocator),
```

Comment:
This creates a backward dependency from a buffer to allocator modules. Do you think it removing `dyn` justifies backward dependency?

Reply:
I would assume `FrameBuffer` and `FrameAllocator` implementation to be close enough that adding a new one won't be much hassle.